src/components/editor/editor-window/tool-bar/utils.ts

```diff
@@ -0,0 +1,128 @@
+import CodeMirror from 'codemirror'
+
+export const updateSelection = (content: string, startPosition: CodeMirror.Position, endPosition: CodeMirror.Position, onContentChange: (content: string) => void, selection: string): void => {
+  const lines = content.split('\n')
+  const newLines = selection.split('\n')
+  const extraLines = newLines.length - 1
```

Comment:
This hole file is very complicated and most of the variable names are confusing. 

Why `lines`? Why does `newLines` contain the selection? Why does `extraLines` contain a number and not line?

Reply:
lines: the lines in the content
newLines: the lines in the selection (the new lines to add)
extraLines: this maybe should be numberOfExtraLines

Reply:
I changed variable names